src/librustc/lint/mod.rs

```diff
@@ -80,7 +80,6 @@ impl Lint {
     /// Returns the `rust::lint::Lint` for a `syntax::early_buffered_lints::BufferedEarlyLintId`.
     pub fn from_parser_lint_id(lint_id: BufferedEarlyLintId) -> &'static Self {
         match lint_id {
-            BufferedEarlyLintId::QuestionMarkMacroSep => QUESTION_MARK_MACRO_SEP,
             BufferedEarlyLintId::IllFormedAttributeInput => ILL_FORMED_ATTRIBUTE_INPUT,
         }
```

Comment:
the whole `match` can now be replaced with just the return argument
```suggestion
        ILL_FORMED_ATTRIBUTE_INPUT
```
Maybe even remove the whole enum as it only contains one variant

Reply:
Indeed, that would make sense. But I wonder if we should do it given that a new enum variant is going to be added in the coming weeks by #62008. What do you think?

Reply:
@ia0 Oh, I wasn't aware of this. Then of course your changes are perfectly fine :smiley: